### DISCUSSION TITLE
ERROR: mkdir '/dev/null/.distcc' failed: Not a directory

### DISCUSSION THREAD

**Discussion by perdakovich** [KSmanis/docker-gentoo-distcc#72] [Q&A] [answered]
I entered the docker container to find the issue why client can't distribute to the host. What I saw is this, and maybe it is the culprit? I'm not sure, I'm just desperate

**Comment by perdakovich** [comment]
![image](https://github.com/user-attachments/assets/b6aefcdf-c25e-4cd0-bdda-4f09fc492841)

**Comment by perdakovich** [comment]
![image](https://github.com/user-attachments/assets/0e241417-ec54-40e4-bd7b-9046be5c132d)

**Comment by KSmanis** [comment, marked as answer]
The default user for the `tcp` image is `distcc`, which has no home directory. Try using `docker exec -itu root gentoo-distcc-tcp bash`, to debug the container as root instead, or you'll be limited to the privileges of the `distcc` user.

**Comment by perdakovich** [comment]
How can I install GCC 14.2 on your container? I need sync with compiler of my client PC

**Comment by KSmanis** [comment]
Only the stable Gentoo tool-chain is currently supported out of the box, you'd have to build your own image if you are interested in different versions, by emerging the appropriate ebuilds.

**Comment by perdakovich** [comment]
![image](https://github.com/user-attachments/assets/dcddb16d-a889-4e7f-9789-87206a3b6417)

How do I change GCC in your container? 

Gentoo Wiki way wouldn't work? 

https://wiki.gentoo.org/wiki/Upgrading_GCC

**Comment by KSmanis** [comment]
The container is not a fully functional Gentoo installation, you'll hit various such issues along the way. The issue above is caused because the portage tree is not checked out by default inside the container. You should consult the Gentoo forum for Gentoo-related questions such as upgrading the system compiler and the repo's Dockerfile for container-related questions. If you are not familiar with Gentoo or Docker, this won't be a trivial process!
